File: salonmanager/frontend/lib/features/gallery_ai/models/gallery_photo.dart
class GalleryPhoto {
  final int id;
  final int salonId;
  final int? albumId;
  final int? customerId;
  final String path;
  final Map<String, dynamic>? variants;
  final String? beforeAfterGroup;
  final DateTime? approvedAt;
  final DateTime? rejectedAt;
  final int createdBy;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? albumTitle;
  final String? albumVisibility;
  final String? customerName;
  final String? creatorName;

  GalleryPhoto({
    required this.id,
    required this.salonId,
    this.albumId,
    this.customerId,
    required this.path,
    this.variants,
    this.beforeAfterGroup,
    this.approvedAt,
    this.rejectedAt,
    required this.createdBy,
    required this.createdAt,
    required this.updatedAt,
    this.albumTitle,
    this.albumVisibility,
    this.customerName,
    this.creatorName,
  });

  factory GalleryPhoto.fromJson(Map<String, dynamic> json) {
    return GalleryPhoto(
      id: json['id'],
      salonId: json['salon_id'],
      albumId: json['album_id'],
      customerId: json['customer_id'],
      path: json['path'],
      variants: json['variants'],
      beforeAfterGroup: json['before_after_group'],
      approvedAt: json['approved_at'] != null ? DateTime.parse(json['approved_at']) : null,
      rejectedAt: json['rejected_at'] != null ? DateTime.parse(json['rejected_at']) : null,
      createdBy: json['created_by'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
      albumTitle: json['album']?['title'],
      albumVisibility: json['album']?['visibility'],
      customerName: json['customer']?['name'],
      creatorName: json['creator']?['name'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'salon_id': salonId,
      'album_id': albumId,
      'customer_id': customerId,
      'path': path,
      'variants': variants,
      'before_after_group': beforeAfterGroup,
      'approved_at': approvedAt?.toIso8601String(),
      'rejected_at': rejectedAt?.toIso8601String(),
      'created_by': createdBy,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }

  bool get isApproved => approvedAt != null && rejectedAt == null;
  bool get isRejected => rejectedAt != null;

  String get imageUrl {
    // This would typically construct the full URL from the path
    // For now, return the path as-is
    return path;
  }

  String? get thumbnailUrl {
    if (variants != null && variants!['thumb'] != null) {
      return variants!['thumb']['path'] ?? variants!['thumb'];
    }
    return imageUrl;
  }
}

File: salonmanager/backend/app/Models/GalleryPhoto.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class GalleryPhoto extends Model
{
    use HasFactory;

    protected $fillable = [
        'salon_id',
        'album_id',
        'customer_id',
        'path',
        'variants',
        'before_after_group',
        'approved_at',
        'rejected_at',
        'created_by',
    ];

    protected function casts(): array
    {
        return [
            'variants' => 'array',
            'approved_at' => 'datetime',
            'rejected_at' => 'datetime',
            'before_after_group' => 'string',
        ];
    }

    public function salon(): BelongsTo
    {
        return $this->belongsTo(Salon::class);
    }

    public function album(): BelongsTo
    {
        return $this->belongsTo(GalleryAlbum::class, 'album_id');
    }

    public function customer(): BelongsTo
    {
        return $this->belongsTo(CustomerProfile::class, 'customer_id');
    }

    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function consents(): HasMany
    {
        return $this->hasMany(GalleryConsent::class, 'photo_id');
    }

    public function likes(): HasMany
    {
        return $this->hasMany(GalleryLike::class, 'photo_id');
    }

    public function favorites(): HasMany
    {
        return $this->hasMany(GalleryFavorite::class, 'photo_id');
    }

    public function beforeAfterPhotos(): HasMany
    {
        return $this->hasMany(GalleryPhoto::class, 'before_after_group', 'before_after_group')
            ->where('id', '!=', $this->id);
    }

    public function isApproved(): bool
    {
        return !is_null($this->approved_at) && is_null($this->rejected_at);
    }

    public function isRejected(): bool
    {
        return !is_null($this->rejected_at);
    }

    public function scopeApproved($query)
    {
        return $query->whereNotNull('approved_at')->whereNull('rejected_at');
    }

    public function scopePublic($query)
    {
        return $query->whereHas('album', function ($q) {
            $q->where('visibility', 'public');
        });
    }
}
